### PR TITLE
Fixed Maxim reloading and zooming

### PR DESCRIPTION
**Pull request by martinlyra**
And added spaghetti a la bolognesa

Addresses #26 and #28

### CODE REVIEW

File: Interstation-Two-WW2/code/game/objects/items.dm [hunk 1]
```diff
@@ -560,32 +560,40 @@ var/list/global/slot_flags_enumeration = list(
...
 
+// An ugly hack called a boolean proc, made it like this to allow special
+// behaviour without big overrides. So special snowflake weapons like the minigun
+// can use zoom without overriding the original zoom proc.
+//	user: user mob
+//	devicename: name of what device you are peering through, set by zoom() in items.dm
+//	silent: boolean controlling whether it should tell the user why they can't zoom in or not
+// I am sorry for creating this abomination -- Irra
+/obj/item/proc/can_zoom(mob/user, var/devicename = src.name, var/silent = 0)
+	if(user.stat || !(istype(user,/mob/living/carbon/human)))
```
Comment (mattroks101) [inline Interstation-Two-WW2/code/game/objects/items.dm]: Use ishuman(user)

File: Interstation-Two-WW2/code/modules/WW2/weapons/guns/stationary.dm [hunk 2]
```diff
@@ -47,13 +55,45 @@
...
+
+// An ugly hack called a boolean proc, made it like this to allow special
+// behaviour without too many overrides. So special snowflake weapons like the minigun
+// can use zoom without overriding the original zoom proc.
+//	user: user mob
+//	devicename: name of what device you are peering through, set by zoom() in items.dm
+//	silent: boolean controlling whether it should tell the user why they can't zoom in or not
+// I am sorry for creating this abomination -- Irra
+/obj/item/weapon/gun/projectile/minigun/can_zoom(mob/user, var/devicename, var/silent)
+	if(user.stat || !(istype(user,/mob/living/carbon/human)))
```
Comment (mattroks101) [inline Interstation-Two-WW2/code/modules/WW2/weapons/guns/stationary.dm]: Same here.

File: Interstation-Two-WW2/code/modules/mob/inventory.dm [hunk 3]
```diff
@@ -143,6 +143,14 @@ var/list/slot_equipment_priority = list( \
 	W.layer = initial(W.layer)
 	W.dropped()
 	return 0
+
+//Checks if mob has one or more empty hands, depending on input param:
+//	both: if 0 (false), it returns true if at least one hand is free, otherwise 1 for checking both
+/mob/proc/has_empty_hand(var/both = 0)
```
Comment (mattroks101) [inline Interstation-Two-WW2/code/modules/mob/inventory.dm]: I thought there was already a proc that checked for something like this but that could just be me going senile.

Comment (martinlyra) [inline Interstation-Two-WW2/code/modules/mob/inventory.dm]: Better safe than sorry

Comment (mattroks101) [inline Interstation-Two-WW2/code/modules/mob/inventory.dm]: Yeah. If I'm not crazy and there is a proc that handles this already it can be replaced later.

Comment (martinlyra) [inline Interstation-Two-WW2/code/modules/mob/inventory.dm]: If it happens to exist in this code, I agree.

**Comment by kachnov** [comment]
if this has been tested, I'll merge

**Comment by martinlyra** [comment]
I have indeed tested this, but feel free to try it on your own.

**Pull request merged**